collector.go

```diff
@@ -191,25 +201,41 @@ func pduToSample(indexOids []int, pdu *gosnmp.SnmpPDU, metric *config.Metric, oi
 		// It's some form of string.
 		t = prometheus.GaugeValue
 		value = 1.0
-		stringType = true
-	}
-
-	labelnames := make([]string, 0, len(labels)+1)
-	labelvalues := make([]string, 0, len(labels)+1)
-	for k, v := range labels {
-		labelnames = append(labelnames, k)
-		labelvalues = append(labelvalues, v)
-	}
-	// For strings we put the value as a label with the same name as the metric.
-	// If the name is already an index, we do not need to set it again.
-	if stringType {
+		// For strings we put the value as a label with the same name as the metric.
+		// If the name is already an index, we do not need to set it again.
 		if _, ok := labels[metric.Name]; !ok {
 			labelnames = append(labelnames, metric.Name)
 			labelvalues = append(labelvalues, pduValueAsString(pdu, metric.Type))
 		}
+		if len(metric.Overrides) > 0 {
+			res = generateOverrideMetrics(metric, pdu.Value.(string), labelnames, labelvalues)
+		}
+	}
+
+	if len(res) == 0 {
+		res = append(res, prometheus.MustNewConstMetric(prometheus.NewDesc(metric.Name, metric.Help, labelnames, nil),
+			t, value, labelvalues...))
+	}
+	return res
+}
+
+func generateOverrideMetrics(metric *config.Metric, pduValue string, labelnames, labelvalues []string) []prometheus.Metric {
+	results := []prometheus.Metric{}
+	for name, strMetricSlice := range metric.Overrides {
+		for _, strMetric := range strMetricSlice {
+			if strMetric.Regex.MatchString(pduValue) {
+				value := strMetric.Regex.ReplaceAllString(pduValue, strMetric.Value)
+				v, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					log.Errorf("Error parsing float64 from value: %v", value)
```

Comment:
This would be a debug log at most, and you need to `continue` when it happens and try the next regex